### ISSUE TITLE
Bring back custom serialization handlers

### ISSUE THREAD

**Issue by pcmoritz** [ray-project/ray#899]
Now that we are registering classes automatically, registering custom serialization handlers is not exposed at the moment. We should bring it back.

**Comment by pcmoritz** [comment]
first step is here: https://github.com/apache/arrow/pull/1019

**Comment by robertnishihara** [comment]
I think what we want is to expose

```python
ray.worker._register_class(cls, pickle=True)
```
although this may actually need to be called from every worker and the driver, it may not be enough to just call it from the driver.

cc @vitchyr

**Comment by vitchyr** [comment]
Yeah, I added this to the beginning of my script and it worked:
```
    serialization.add_class_to_whitelist(
        ClassThatYouWantToUseCustomSerialization,
        utils.random_string(),
        pickle=True,
    )
```
I haven't tested it on multiple workers, but I don't see why it shouldn't work.

**Comment by robertnishihara** [comment]
Well I think this works if `ClassThatYouWantToUseCustomSerialization` is being serialized on the driver and then deserialized on the workers. If it is instead being serialized on the workers and deserialized on the driver (or on another worker) then it's not enough I think.

**Comment by robertnishihara** [comment]
Fixed by #1147.
